test/browser_tests/spec_suites/organisms/header.js

```diff
@@ -135,11 +134,31 @@ describe( 'Header', function() {
               expect( _dom.overlay.isDisplayed() ).toBe( true );
             } );
         } );
+
+        describe( 'then click search', function() {
+          it( 'should show the search and hide menu', function() {
+            browser.driver.actions().click( _dom.megaMenuTrigger ).perform();
+            browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
+            browser.driver.actions().click( _dom.globalSearchTrigger ).perform()
+              .then( function() {
+                expect( _dom.globalSearch.isDisplayed() ).toBe( true );
+
+              } );
+          } );
+        } );
+      } );
+
+      describe( 'click search, then click mega menu', function() {
+        it( 'should show the megamenu and hide search', function() {
+          browser.driver.actions().click( _dom.globalSearchTrigger ).perform();
+          browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
```

Comment:
But you're opening the search, waiting for the menu, then opening the menu. That should be open search, wait for search, then open the menu. That's why I'm not so sure the wait is doing anything here and why it might not be doing anything elsewhere. It should be timing out, because the menu was never triggered and therefore wouldn't ever return that it's content is being displayed.